decentralizediso20022/src/main/java/com/centiglobe/decentralizediso20022/util/HTTPSCustomTruststore.java

```diff
@@ -45,7 +45,8 @@ public static void configureTruststore(HttpsURLConnection connection, String tru
         CertPathBuilder cpb = CertPathBuilder.getInstance("PKIX");
         PKIXRevocationChecker rc = (PKIXRevocationChecker) cpb.getRevocationChecker();
         rc.setOptions(EnumSet.of(PKIXRevocationChecker.Option.PREFER_CRLS, // prefer CLR over OCSP
-                PKIXRevocationChecker.Option.ONLY_END_ENTITY, PKIXRevocationChecker.Option.NO_FALLBACK));
+                PKIXRevocationChecker.Option.SOFT_FAIL, PKIXRevocationChecker.Option.NO_FALLBACK));
+                // TODO: Change to PKIXRevocationChecker.Option.ONLY_END_ENTITY
```

Comment:
We shouldn't soft fail. Why was this changed? Is it something with the certificates used in testing?
From JavaDocs:
_SOFT_FAIL | Allow revocation check to succeed if the revocation status cannot be determined for one of the following reasons: The CRL or OCSP response cannot be obtained because of a network error._
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/cert/PKIXRevocationChecker.Option.html

This means that we might consider a revoked certificate valid if we use soft fail.

Reply:
It was because it could not detect the revocation status of the server's (external service's) certificate, probably because of it being just a demo. Is this something that could be resolved in a future PR? I added a TODO to signify that it should be changed.

Reply:
I think we should also add a note on removing soft fail.

Reply:
Do you mean an issue?

Reply:
I was thinking of just a comment in the code so that we know that soft fail should also be removed.

Reply:
I appended removal of `SOFT_FAIL` to the comment.